File: CS/Styles/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
#region #usings
using DevExpress.XtraRichEdit.API.Native;
#endregion #usings

namespace Styles
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();

            richEditControl1.LoadDocument("Styles.docx");
        }



private void btnCharacterStyle_Click(object sender, EventArgs e)
{
    #region #cstyle
    CharacterStyle cStyle = richEditControl1.Document.CharacterStyles["MyCharStyle"];
    if (cStyle == null)
    {
        cStyle = richEditControl1.Document.CharacterStyles.CreateNew();
        cStyle.Name = "MyCharStyle";
        cStyle.Parent = richEditControl1.Document.CharacterStyles["Default Paragraph Font"];
        cStyle.ForeColor = Color.DarkOrange;
        cStyle.Strikeout = StrikeoutType.Double;
        cStyle.FontName = "Verdana";
        richEditControl1.Document.CharacterStyles.Add(cStyle);
    }
    DocumentRange r = richEditControl1.Document.Selection;
    CharacterProperties charProps =
        richEditControl1.Document.BeginUpdateCharacters(r);
    charProps.Style = cStyle;
    richEditControl1.Document.EndUpdateCharacters(charProps);
    #endregion #cstyle
}

private void btnParagraphStyle_Click(object sender, EventArgs e)
{
    #region #pstyle
    ParagraphStyle pStyle = richEditControl1.Document.ParagraphStyles["MyParStyle"];
    if (pStyle == null)
    {
        pStyle = richEditControl1.Document.ParagraphStyles.CreateNew();
        pStyle.Name = "MyParStyle";
        pStyle.LineSpacingType = ParagraphLineSpacing.Double;
        pStyle.Alignment = ParagraphAlignment.Center;
        richEditControl1.Document.ParagraphStyles.Add(pStyle);
    }
    DocumentRange r = richEditControl1.Document.Selection;
    ParagraphProperties parProps =
        richEditControl1.Document.BeginUpdateParagraphs(r);
    parProps.Style = pStyle;
    richEditControl1.Document.EndUpdateParagraphs(parProps);
    #endregion #pstyle
}

private void btnLinkedStyle_Click(object sender, EventArgs e)
{
    #region #lstyle
    ParagraphStyle linkedStyle = richEditControl1.Document.ParagraphStyles["MyLinkedParStyle"];
    if (linkedStyle == null)
    {
        linkedStyle = richEditControl1.Document.ParagraphStyles.CreateNew();
        linkedStyle.Name = "MyLinkedParStyle";
        linkedStyle.LineSpacingType = ParagraphLineSpacing.Single;
        linkedStyle.Alignment = ParagraphAlignment.Justify;
        richEditControl1.Document.ParagraphStyles.Add(linkedStyle);
    }
    CharacterStyle lcstyle = richEditControl1.Document.CharacterStyles.CreateNew();
    lcstyle.Name = "MyLinkedCharStyle";
    richEditControl1.Document.CharacterStyles.Add(lcstyle);
    lcstyle.LinkedStyle = linkedStyle;

    lcstyle.ForeColor = Color.DarkGreen;
    lcstyle.Strikeout = StrikeoutType.Single;
    lcstyle.FontSize = 24;
    #endregion #lstyle
}
}
}

File: VB/Styles/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
#Region "#usings"
Imports DevExpress.XtraRichEdit.API.Native
#End Region ' #usings

Namespace Styles
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()

            richEditControl1.LoadDocument("Styles.docx")
        End Sub



Private Sub btnCharacterStyle_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCharStyle.Click
'    #Region "#cstyle"
    Dim cStyle As CharacterStyle = richEditControl1.Document.CharacterStyles("MyCharStyle")
    If cStyle Is Nothing Then
        cStyle = richEditControl1.Document.CharacterStyles.CreateNew()
        cStyle.Name = "MyCharStyle"
        cStyle.Parent = richEditControl1.Document.CharacterStyles("Default Paragraph Font")
        cStyle.ForeColor = Color.DarkOrange
        cStyle.Strikeout = StrikeoutType.Double
        cStyle.FontName = "Verdana"
        richEditControl1.Document.CharacterStyles.Add(cStyle)
    End If
    Dim r As DocumentRange = richEditControl1.Document.Selection
    Dim charProps As CharacterProperties = richEditControl1.Document.BeginUpdateCharacters(r)
    charProps.Style = cStyle
    richEditControl1.Document.EndUpdateCharacters(charProps)
'    #End Region ' #cstyle
End Sub

Private Sub btnParagraphStyle_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnParStyle.Click
'    #Region "#pstyle"
    Dim pStyle As ParagraphStyle = richEditControl1.Document.ParagraphStyles("MyParStyle")
    If pStyle Is Nothing Then
        pStyle = richEditControl1.Document.ParagraphStyles.CreateNew()
        pStyle.Name = "MyParStyle"
        pStyle.LineSpacingType = ParagraphLineSpacing.Double
        pStyle.Alignment = ParagraphAlignment.Center
        richEditControl1.Document.ParagraphStyles.Add(pStyle)
    End If
    Dim r As DocumentRange = richEditControl1.Document.Selection
    Dim parProps As ParagraphProperties = richEditControl1.Document.BeginUpdateParagraphs(r)
    parProps.Style = pStyle
    richEditControl1.Document.EndUpdateParagraphs(parProps)
'    #End Region ' #pstyle
End Sub

Private Sub btnLinkedStyle_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnLink.Click
'    #Region "#lstyle"
    Dim linkedStyle As ParagraphStyle = richEditControl1.Document.ParagraphStyles("MyLinkedParStyle")
    If linkedStyle Is Nothing Then
        linkedStyle = richEditControl1.Document.ParagraphStyles.CreateNew()
        linkedStyle.Name = "MyLinkedParStyle"
        linkedStyle.LineSpacingType = ParagraphLineSpacing.Single
        linkedStyle.Alignment = ParagraphAlignment.Justify
        richEditControl1.Document.ParagraphStyles.Add(linkedStyle)
    End If
    Dim lcstyle As CharacterStyle = richEditControl1.Document.CharacterStyles.CreateNew()
    lcstyle.Name = "MyLinkedCharStyle"
    richEditControl1.Document.CharacterStyles.Add(lcstyle)
    lcstyle.LinkedStyle = linkedStyle

    lcstyle.ForeColor = Color.DarkGreen
    lcstyle.Strikeout = StrikeoutType.Single
    lcstyle.FontSize = 24
'    #End Region ' #lstyle
End Sub
    End Class
End Namespace